lib/logger.rb

```diff
@@ -540,67 +590,62 @@ def initialize(logdev, shift_age = 0, shift_size = 1048576, level: DEBUG,
     end
   end
 
-  #
-  # :call-seq:
-  #   Logger#reopen
-  #   Logger#reopen(logdev)
-  #
-  # === Args
-  #
-  # +logdev+::
-  #   The log device.  This is a filename (String) or IO object (typically
-  #   +STDOUT+, +STDERR+, or an open file).  reopen the same filename if
-  #   it is +nil+, do nothing for IO.  Default is +nil+.
-  #
-  # === Description
-  #
-  # Reopen a log device.
+  # Sets the logger's output stream:
+  #
+  # - If +logdev+ is +nil+, reopens the already-established output stream.
+  # - If +logdev+ is a filepath, opens the indicated file for append.
+  # - If +logdev+ is an IO stream
+  #   (usually <tt>$stdout</tt>, <tt>$stderr</tt>, or an open File object),
+  #   opens the stream for append.
+  #
+  # Example:
+  #
+  #   logger = Logger.new('t.log')
+  #   logger.add(3)
+  #   logger.add(3)
+  #   logger.close
+  #   logger.add(3) # Prints 'log writing failed. closed stream'
+  #   logger.reopen
+  #   logger.add(3)
+  #   logger.add(3)
+  #   logger.close
+  #   File.readlines('t.log')
+  #   # =>
+  #   ["# Logfile created on 2022-05-12 14:21:19 -0500 by logger.rb/v1.5.0\n",
+  #    "E, [2022-05-12T14:21:27.596726 #22428] ERROR -- : nil\n",
+  #    "E, [2022-05-12T14:21:30.649644 #22428] ERROR -- : nil\n",
+  #    "E, [2022-05-12T14:23:01.353310 #22428] ERROR -- : nil\n",
+  #    "E, [2022-05-12T14:23:05.847241 #22428] ERROR -- : nil\n"]
```

Comment:
Rather than calling `logger.add(3)` repeatedly, which is confusing and difficult to follow for the reader, maybe we should have unique messages for each of them too. For example:

```suggestion
  #   logger.add(Logger::ERROR, 'one')
  #   logger.close
  #   logger.add(Logger::ERROR, 'two') # Prints 'log writing failed. closed stream'
  #   logger.reopen
  #   logger.add(Logger::ERROR, 'three')
  #   logger.close
  #   File.readlines('t.log')
  #   # =>
  #   # ["# Logfile created on 2022-05-12 14:21:19 -0500 by logger.rb/v1.5.0\n",
  #   #  "E, [2022-05-12T14:21:27.596726 #22428] ERROR -- : one\n",
  #   #  "E, [2022-05-12T14:23:05.847241 #22428] ERROR -- : three\n"]
```